test-common/src/main/java/org/triplea/test/common/swing/SwingComponentWrapper.java

```diff
@@ -0,0 +1,58 @@
+package org.triplea.test.common.swing;
+
+import java.awt.Component;
+import java.awt.Container;
+import java.util.Arrays;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import lombok.Value;
+
+/**
+ * A decorator or wrapper for a javax.swing Component that adds methods useful for testing.
+ */
+@Value(staticConstructor = "of")
+public class SwingComponentWrapper {
+
+  @Nonnull
+  private final Component component;
+
+  /**
+   * Recursively searches the wrapped component for a swing component whose name matches the parameter,
+   * any such component found is then returned and casted to the type specified by the second parameter.
+   * *
+   * 
+   * @param childName Component name to find, searching current component or any of its children.
+   * @param classType The component type to return.
+   * @param <T> The expected type of the component that we are trying to find.
+   * @return A matching component casted to the specified type or an assertion error.
+   * @throws ClassCastException Thrown if the matched component has a type that is different from the expected type.
+   * @throws AssertionError Thrown when failing to match any components by name.
+   */
+  public <T> T findChildByName(final String childName, final Class<T> classType) {
```

Comment:
Would suggest naming this pair of methods one of the following:

1. `findChildByNameOrThrow()`, `findChildByName()`
2. `findChildByName()`, `tryFindChildByName()`

Reply:
:smiley: good to see the suggestions; valuable feedback.

0fcf77df0 I think is the 'correct' way to address this.


`<storytime>`

There is a bit of a history why the test exposed method is there. That was the original public API. It then turned out because this is a test context, *every* client (each client being a unit test) can and will want to handle the item not found event with an error and not deal with the optional otherwise.

I was losing steam when updating this, with renewed effort it seems clear the test should be updated to verify the public API and check for exceptions. I was attempting to combine the two methods which then made me realize that we have the second method not for testing only, but also to allow for a recursive call. See the commit comment for more details. I think this lead to the better name 'findChildByNameRecursive'. With that the API makes more sense.

Otherwise initially I was approaching this problem to keep `findChildByName()` fixed as that was public API and any additional naming is a bit excessive when clients are using this API. The question then was what to name the internal method, my first reaction was then to avoid the problem by combining the two methods, and hence the realization for a better name...    
`</storytime>`